src/jit/Compiler.h

```diff
@@ -202,30 +203,63 @@ class Instruction : public InstructionListItem {
         return reinterpret_cast<BrTableInstruction*>(this);
     }
 
+    uint8_t tmpReg(size_t n)
+    {
+        ASSERT(n < sizeof(m_tmpRegs));
+        return m_tmpRegs[n];
+    }
+
+    void setTmpReg(size_t n, uint8_t value)
+    {
+        ASSERT(n < sizeof(m_tmpRegs));
+        m_tmpRegs[n] = value;
+    }
+
+    inline void setTmpReg2(uint8_t value1, uint8_t value2)
+    {
+        m_tmpRegs[0] = value1;
+        m_tmpRegs[1] = value2;
+    }
+
+    inline void setTmpReg3(uint8_t value1, uint8_t value2, uint8_t value3)
+    {
+        setTmpReg2(value1, value2);
+        m_tmpRegs[2] = value3;
+    }
+
+    inline void setTmpReg4(uint8_t value1, uint8_t value2, uint8_t value3, uint8_t value4)
+    {
+        setTmpReg3(value1, value2, value3);
+        m_tmpRegs[3] = value4;
+    }
+
 protected:
     explicit Instruction(ByteCode* byteCode, Group group, ByteCode::Opcode opcode, uint32_t paramCount, Operand* operands, InstructionListItem* prev)
         : InstructionListItem(group, prev)
         , m_byteCode(byteCode)
+        , m_operands(operands)
         , m_opcode(opcode)
         , m_paramCount(paramCount)
-        , m_operands(operands)
     {
+        memset(m_tmpRegs, 0, sizeof(m_tmpRegs));
     }
 
     explicit Instruction(ByteCode* byteCode, Group group, ByteCode::Opcode opcode, InstructionListItem* prev)
         : InstructionListItem(group, prev)
         , m_byteCode(byteCode)
+        , m_operands(nullptr)
         , m_opcode(opcode)
         , m_paramCount(0)
-        , m_operands(nullptr)
     {
+        memset(m_tmpRegs, 0, sizeof(m_tmpRegs));
     }
 
 private:
     ByteCode* m_byteCode;
+    Operand* m_operands;
     ByteCode::Opcode m_opcode;
     uint32_t m_paramCount;
-    Operand* m_operands;
+    uint8_t m_tmpRegs[4];
```

Comment:
Each `Instruction` has up to 4 temp-registers now.
Is this sufficient to handle all operations?

Reply:
Currently 4 should be enough. However, I am thinking about a different descriptor system for temporaries and arguments, which stores the data in a 32 bit value. This is replaced by the maximum 4 temporary values (using an union). This code will probably not be the final code, but we should start somewhere.

Reply:
4 temporary registers allocated in a Union form would be great too